django/db/backends/base/schema.py

```diff
@@ -538,6 +538,14 @@ def alter_field(self, model, old_field, new_field, strict=False):
         If `strict` is True, raise errors if the old column does not match
         `old_field` precisely.
         """
+        # Don't alter if only field name has changed.
+        if (
+            old_field.name != new_field.name and
+            old_field.column == new_field.column and
+            old_field.deconstruct()[1:] == new_field.deconstruct()[1:]
+        ):
+            return
```

Comment:
I think we should create a hook similar to the `_field_should_be_indexed()`, that will allow 3rd-party backend to adjust this behavior, e.g.
```python
    def _field_should_be_altered(self, old_field, new_field):
        # Don't alter when changing only a field name.
        return (
            old_field.column != new_field.column or
            old_field.deconstruct()[1:] != new_field.deconstruct()[1:]
        )
```

Reply:
Done, it is cleaner. :+1: